core/meta/model-reference.js

```diff
@@ -0,0 +1,79 @@
+var Promise = require("../promise").Promise,
+    RemoteReference = require("./remote-reference").RemoteReference,
+    ModelModule = require("./model");
+
+exports.ModelReference = RemoteReference.specialize({
+
+    constructor: {
+        value: function ModelReference() {
+            this.superForValue("constructor")();
+        }
+    },
+
+    /**
+     * The identifier is the name of the object model and is used to make the
+     * serialization of object models more readable.
+     * @readonly
+     * @type {string}
+     */
+    identifier: {
+        get: function () {
+            if (!this._reference) {
+                this._reference = this.referenceFromValue(this._value);
+            }
+            return [
+                "model",
+                this._reference.modelName.toLowerCase(),
+                "reference"
+            ].join("_");
+        }
+    },
+
+    valueFromReference: {
+        value: function (references, require) {
+            var modelName = references.modelName,
+                modelModuleId = references.modelModuleId,
+                model = ModelModule.Model.group.modelForName(modelName);
+                // deferredObjectModel,
+                // targetRequire,
+                // slashIndex,
+                // prefix,
+                // mappings;
+
+            return model ?  Promise.resolve(model) :
+                            Promise.reject(new Error("Error cannot find Object Model " + modelModuleId));
+            //         if (objectModel) {
```

Comment:
Not sure.  The commented part always failed because there was an undefined method in it.  See TODO.  I left it in there in case it something we want to fix.